Source/OxyPlot/Utilities/EnumerableExtensions.cs

```diff
@@ -42,5 +44,185 @@ private static IEnumerable<TSource> CreateReverseIterator<TSource>(IList<TSource
                 yield return source[i];
             }
         }
+
+        /// <summary>
+        /// Calculates the percentile for the list of values
+        /// </summary>
+        /// <param name="values">The list of values</param>
+        /// <param name="percentile">The percentile to be calculated.</param>
+        /// <returns>The percentile of the list</returns>
+        public static double PercentilePos(this IEnumerable<double> values, double percentile)
```

Comment:
This should not be in the OxyPlot core project. Belongs to a 3rd party library or in the example library.